### PR TITLE
 feat: add a sources option allowing to bypass fs operations

### LINKED ISSUE #33: Avoid hitting the file system
I'd like to use this module in Jest. But this module tries to look at the filesystem and read sourcemaps on its own, which breaks down. I'd love an alternative API that allows me to pass in:

- raw source code (from "disk")
- the source code that was executed/instrumented
- the source map for this

I cobbled this together quickly, which allowed me to get a PoC working in Jest: https://github.com/SimenB/v8-to-istanbul/commit/6d1560b890f64b05e476e64719e3ffd26ddf357c

### PR DESCRIPTION
**Pull request by SimenB**
This is a first, quick, pass for #33. It does not have docs or tests as I'd like to get some feedback on it if it's the correct approach, or if we should have different `load` methods depending on if we want to read from fs or not

Fixes #33

(includes #35)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/// <reference types="node" />
```
Comment (bcoe) [inline index.d.ts]: this is the format for referencing another type, in this case `@types/node`? new to me.

Comment (SimenB) [reply index.d.ts]: It's so that TS knows that `inspector` comes from `node` - it's not a separate module

**Comment by SimenB** [comment]
Definitely needs a test or two! I'll come back to it once we've settled on an API. This one feels a bit brute force.

I think maybe some of this sourcemap logic could move to `istanbul-lib-source-maps`? Maybe not

File: lib/v8-to-istanbul.js [hunk 2]
```diff
@@ -27,11 +28,13 @@ module.exports = class V8ToIstanbul {
     this.sourceTranspiled = undefined
   }
   async load () {
-    const rawSource = readFileSync(this.path, 'utf8')
-    // if we find a source-map (either inline, or a .map file) we load
-    // both the transpiled and original source, both of which are used during
-    // the backflips we perform to remap absolute to relative positions.
-    const rawSourceMap = convertSourceMap.fromSource(rawSource) || convertSourceMap.fromMapFileSource(rawSource, dirname(this.path))
+    const rawSource = this.sources.source || readFileSync(this.path, 'utf8')
```
Comment (shinnn) [inline lib/v8-to-istanbul.js]: When `this.sources.source` is a zero-length string, it still tries to read an actual file. I guess you meant

```js
this.sources.source === undefined ? readFileSync(this.path, 'utf8') : this.sources.source
```

Comment (SimenB) [reply lib/v8-to-istanbul.js]: good call!

FWIW, I'm still not happy with the branching logic I introduced in this PR, I'd prefer an explicit `.loadFromFs` and `.loadFromMemory` or something API. Haven't had time to dig further into this yet though

Comment (shinnn) [reply lib/v8-to-istanbul.js]: 👍 Separate APIs really sound good to me.

**Pull request merged**